src/wtforms/form.py

```diff
@@ -143,11 +141,7 @@ def data(self):
 
     @property
     def errors(self):
-        if self._errors is None:
-            self._errors = dict(
-                (name, f.errors) for name, f in iteritems(self._fields) if f.errors
-            )
-        return self._errors
+        return dict((name, f.errors) for name, f in iteritems(self._fields) if f.errors)
```

Comment:
This should be a dict comprehension and `iteritems` is no longer needed. `{name: f.errors for name, f in self._fields.items() if f.errors}`